build.gradle.kts

```diff
@@ -5,323 +5,325 @@ import kotlin.collections.ArrayList
+  inputs.dir("$buildDir/libs")
+  inputs.dir("$projectDir/support/jpackage/linux")
+  outputs.file(ext.get("rpmFilename") as String)
+  doLast {
+    if (OperatingSystem.current().isLinux) {
+      val parameters = ArrayList<String>(ext.get("sharedParameters") as ArrayList<String>)
+      parameters.addAll(ext.get("linuxParameters") as ArrayList<String>)
+      parameters.addAll(Arrays.asList(
+          "--type", "rpm"
+      ))
+      val processBuilder2 = ProcessBuilder()
+      processBuilder2.command(parameters)
+      val process2 = processBuilder2.start()
+      if (process2.waitFor() != 0) {
+        throw GradleException("Error while creating rpm package")
       }
-   }
+    }
+  }
 }
 
 tasks.register("createMsi") {
-   group = "build"
-   description = "Makes the Windows platform specific package"
-   dependsOn("shadowJar", "createDistDir")
-   inputs.dir("$buildDir/libs")
-   inputs.dir("$projectDir/support/jpackage/windows")
-   outputs.file(ext.get("msiFilename") as String)
-   doLast {
-      if (OperatingSystem.current().isWindows) {
-         val parameters = ArrayList<String>(ext.get("sharedParameters") as ArrayList<String>)
-         parameters.addAll(Arrays.asList(
-            "--name", project.name,
-            "--file-associations", "$projectDir/support/jpackage/windows/file.jpackage",
-            "--icon", "$projectDir/support/jpackage/windows/Logisim-evolution.ico",
-            "--type", "msi",
-            "--win-menu-group", "logisim",
-            "--win-shortcut",
-            "--win-dir-chooser",
-            "--win-menu"
-         ))
-         val processBuilder1 = ProcessBuilder()
-         processBuilder1.command(parameters)
-         val process1 = processBuilder1.start()
-         if (process1.waitFor() != 0) {
-            throw GradleException("Error while creating msi package")
-         }
+  group = "build"
+  description = "Makes the Windows platform specific package"
+  dependsOn("shadowJar", "createDistDir")
+  inputs.dir("$buildDir/libs")
+  inputs.dir("$projectDir/support/jpackage/windows")
+  outputs.file(ext.get("msiFilename") as String)
+  doLast {
+    if (OperatingSystem.current().isWindows) {
+      val parameters = ArrayList<String>(ext.get("sharedParameters") as ArrayList<String>)
+      parameters.addAll(Arrays.asList(
+          "--name", project.name,
+          "--file-associations", "$projectDir/support/jpackage/windows/file.jpackage",
+          "--icon", "$projectDir/support/jpackage/windows/Logisim-evolution.ico",
+          "--type", "msi",
+          "--win-menu-group", "logisim",
+          "--win-shortcut",
+          "--win-dir-chooser",
+          "--win-menu"
+      ))
+      val processBuilder1 = ProcessBuilder()
+      processBuilder1.command(parameters)
+      val process1 = processBuilder1.start()
+      if (process1.waitFor() != 0) {
+        throw GradleException("Error while creating msi package")
       }
-   }
+    }
+  }
 }
 
 tasks.register("createApp") {
```

Comment:
is `createApp` task useful for anything (ignorant question)?

Reply:
I can't speak for everyone; but I use createApp more than any other task. It builds the Mac .app without wasting time building the .dmg file. When we only had the target 'jpackage', I would use that to build the .dmg file, open the .dmg, drag out the .app, then open it. With createApp, I can type './gradlew createApp' and then 'open build/dist/Logisim-evolution.app' to open it. That is much faster (especially with tab completes). That is why I started building the createXxx tasks in the first place. Before jpackage was included with Java, we had an old createApp task that built the Mac .app. That is were the name comes from.

Reply:
If you were actually asking if we could just delete createApp, no. That would break createDmg also, since createDmg depends on createApp. All createDmg actually does is take the .app created by createApp and puts it into a disk image.

Reply:
Ok, thanks for clarification.